Add and modify permissions tables (#4936)
fleetdm/fleet: docs/Using-Fleet/Permissions.md

The following table depicts various permissions levels for each role.

| **Action**                                           | Observer | Maintainer | Admin |
| ---------------------------------------------------- | -------- | ---------- | ----- |
| View activity                                        | ✅       | ✅         | ✅    |
| Browse all hosts                                     | ✅       | ✅         | ✅    |
| Filter hosts using labels                            | ✅       | ✅         | ✅    |
| Target hosts using labels                            | ✅       | ✅         | ✅    |
| Enroll hosts                                         |          | ✅         | ✅    |
| Delete hosts                                         |          | ✅         | ✅    |
| Transfer hosts between teams\*                       |          | ✅         | ✅    |

> | Create, edit, and Delete labels                      |          | ✅         | ✅    |

Reviewer: ```suggestion
| Create, edit, and delete labels                      |          | ✅         | ✅    |
```

Sentence casing on "delete."

> | Create, edit, and Delete labels                      |          | ✅         | ✅    |

Reviewer: ```suggestion
| Create, edit, and delete labels                      |          | ✅         | ✅    |
```

Sentence casing on "delete."

> | Browse all software                                  | ✅       | ✅         | ✅    |

Reviewer: ```suggestion
| View all software                                  | ✅       | ✅         | ✅    |
```

| Filter software by vulnerabilities                   | ✅       | ✅         | ✅    |
| Filter hosts by software                             | ✅       | ✅         | ✅    |
| Filter software by team*                             | ✅       | ✅         | ✅    |
| Manage software by vulnerability automations         |          | ✅         | ✅    |

> | Run saved queries as live queries against all hosts  | ✅       | ✅         | ✅    |

Reviewer: ```suggestion
| Run only designated, _observer can run_ ,queries as live queries against all hosts  | ✅       | ✅         | ✅    |
```

I suggest specifying the observer can run functionality here.

> | Run saved queries as live queries against all hosts  | ✅       | ✅         | ✅    |

Reviewer: ```suggestion
| Run only designated, _observer can run_ ,queries as live queries against all hosts  | ✅       | ✅         | ✅    |
```

I suggest specifying the observer can run functionality here.

> | Run custom queries as live queries against all hosts |          | ✅         | ✅    |

Reviewer: ```suggestion
| Run any query as live query against all hosts |          | ✅         | ✅    |
```
I suggest updating this to "any query" so that this line is clearly distinguished from the above "observer can run" line.

> | Run custom queries as live queries against all hosts |          | ✅         | ✅    |

Reviewer: ```suggestion
| Run any query as live query against all hosts |          | ✅         | ✅    |
```
I suggest updating this to "any query" so that this line is clearly distinguished from the above "observer can run" line.

| Create, edit, and delete queries                     |          | ✅         | ✅    |

> | Create, edit, and delete scheduled queries for all hosts |          | ✅         | ✅    |

Reviewer: ```suggestion
| Create, edit, view, and delete scheduled queries for all hosts |          | ✅         | ✅    |
```

Reviewer: TODO: Add a new "View all queries" line with checks for observer, maintainer, and admins.

| Create, edit, and delete scheduled queries for all hosts assigned to a team\*  |          | ✅         | ✅    |

> | Create, edit, and delete packs                       |          | ✅         | ✅    |

Reviewer: ```suggestion
| Create, edit, view, and delete packs                       |          | ✅         | ✅    |
```

> | Browse all policies                                  | ✅       | ✅         | ✅    |

Reviewer: ```suggestion
| View all policies                                  | ✅       | ✅         | ✅    |
```
What do you think about using the word "View" instead of "Browse?" This seems to be more consistent with other applications' permissions docs (ex. [GitLab](https://docs.gitlab.com/ee/user/permissions.html)).

If you agree, can you please make sure that we have "View" permissions included for other entities? (queries, users, etc.)

Author: I like view instead of Browse, I'll make the changes!

> | Browse all policies                                  | ✅       | ✅         | ✅    |

Reviewer: ```suggestion
| View all policies                                  | ✅       | ✅         | ✅    |
```
What do you think about using the word "View" instead of "Browse?" This seems to be more consistent with other applications' permissions docs (ex. [GitLab](https://docs.gitlab.com/ee/user/permissions.html)).

If you agree, can you please make sure that we have "View" permissions included for other entities? (queries, users, etc.)

| Filter hosts using policies                          | ✅       | ✅         | ✅    |
| Create, edit, and delete policies for all hosts      |          | ✅         | ✅    |
| Create, edit, and delete policies for all hosts assigned to team\*     |          | ✅         | ✅    |
| Manage failing policy automations for all hosts      |          | ✅         | ✅    |

> | Create, edit, and delete users                       |          |            | ✅    |

Reviewer: ```suggestion
| Create, edit, view, and delete users                       |          |            | ✅    |
```

> The following table depicts various permissions levels in a team.

Reviewer: ```suggestion
The following table depicts permissions levels in a team.

All hosts, software, policies, etc. are exclusive to one team.
```
TODO workshop this. We're trying to communicate that everything is this table is only accessible if the user is a member of that team.

> | **Action**                                                   | Observer | Maintainer | Admin   |

Reviewer: ```suggestion
| **Action**                                                   | Team observer | Team maintainer | Team admin   |
```

| ------------------------------------------------------------ | -------- | ---------- | ------- |

> | Browse hosts assigned to team                                | ✅       | ✅         | ✅       |

Reviewer: ```suggestion
| View hosts                               | ✅       | ✅         | ✅       |
```
